src/protocol/codec/message.c

```diff
@@ -102,6 +102,7 @@ z_result_t _z_timestamp_decode(_z_timestamp_t *ts, _z_zbuf_t *zbf) {
     _Z_DEBUG("Decoding _TIMESTAMP");
     z_result_t ret = _Z_RES_OK;
 
+    ts->valid = true;
```

Comment:
What if decoding below fails, should timestamp be still marked as valid. I think it is better to set at the end to (res == _Z_RES_OK)